modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp

```diff
@@ -309,207 +345,381 @@ void run_yuv2rgb_impl(uchar out[], const uchar in[], int width, const float coef
     }
 }
 
-//---------------------
+//-------------------------
 //
-// Fluid kernels: Sobel
+// Fluid kernels: sepFilter
 //
-//---------------------
+//-------------------------
 
-// Sobel 3x3: vertical pass
-template<bool noscale, typename DST>
-static void run_sobel3x3_vert(DST out[], int length, const float ky[],
-                float scale, float delta, const int r[], float *buf[])
+#if CV_SIMD
+// this variant not using buf[] appears 15% faster than reference any-2-float code below
+template<bool noscale, typename SRC>
+static void run_sepfilter3x3_any2float(float out[], const SRC *in[], int width, int chan,
+                                       const float kx[], const float ky[], int border,
+                                       float scale, float delta)
 {
-    float ky0 = ky[0],
-          ky1 = ky[1],
-          ky2 = ky[2];
+    const int length = width * chan;
```

Comment:
What is about cases `length` < `nlanes` ?

Reply:
The *reference* code would process such small cases.